ftw/publisher/sender/tests/test_example_workflow_constraint_definition.py

```diff
@@ -19,112 +16,140 @@
 EXAMPLE_WF_REVISION = 'publisher-example-workflow--STATUS--revision'
 
 
-class TestExampleWFConstraintDefinition(TestCase):
+class TestExampleWFConstraintDefinition(FunctionalTestCase):
 
-    layer = PUBLISHER_SENDER_FUNCTIONAL_TESTING
 
     def setUp(self):
-        self.portal = self.layer['portal']
-        self.request = self.layer['request']
-
-        setRoles(self.portal, TEST_USER_ID, ['Manager', 'Reviewer'])
-        login(self.portal, TEST_USER_NAME)
+        super(TestExampleWFConstraintDefinition, self).setUp()
+        self.grant('Manager')
 
         wftool = getToolByName(self.portal, 'portal_workflow')
         wftool.setChainForPortalTypes(['Document', 'Folder', 'ContentPage'],
                                       EXAMPLE_WF_ID)
 
         transaction.commit()
-
-    def test_warning_on_submit_when_parent_is_not_published(self):
+        
+    @browsing
+    def test_warning_on_submit_when_parent_is_not_published(self, browser):
         folder = create(Builder('folder')
                         .in_state(EXAMPLE_WF_INTERNAL))
         page = create(Builder('page')
                       .within(folder)
                       .in_state(EXAMPLE_WF_INTERNAL))
 
-        Plone().login().visit(page)
+        browser.login().visit(page)
         Workflow().do_transition('submit')
 
-        Workflow().assert_portal_message(
-            'warning', 'The parent object needs to be published first.')
+        self.assertEqual(
+            [
+                'The parent object needs to be published first.'
+            ],
+            statusmessages.warning_messages(),
```

Comment:
Use [`statusmessages.assert_message`](http://ftwtestbrowser.readthedocs.io/en/1.23.1/api_pages.html#ftw.testbrowser.pages.statusmessages.assert_message) instead.
I'd also assert that there are no error messages:
```python
statusmessages.assert_no_error_messages()
statusmessages.assert_message('The parent object needs to be published first.')
```